File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.js
function calculate(r, w) {
    return Math.sqrt(r ** 2 - w ** 2);
}

function solution(r1, r2) {
    let answer = 0;
    
    for (let w = 1; w < r2; w++) {
        const h1 = calculate(r1, w);
        const h2 = calculate(r2, w);
        
        if (w >= r1) {
            answer += Math.floor(h2) + 1;
            continue;
        }
        
        answer += Math.floor(h2) - Math.floor(h1);
        
        if (Number.isInteger(h1)) {
            answer += 1;
        }
    }
    
    return 4 * (answer + 1);
}

File: 프로그래머스/2/181187. 두 원 사이의 정수 쌍/두 원 사이의 정수 쌍.java
public class Solution {
    
    private double calculate(long r, long w) {
        return Math.sqrt(Math.pow(r, 2) - Math.pow(w, 2));
    }

    public long solution(int r1, int r2) {
        long answer = 0;

        for (int w = 1; w < r2; w++) {
            double h1 = calculate(r1, w);
            double h2 = calculate(r2, w);

            if (w >= r1) {
                answer += (long) Math.floor(h2) + 1;
                continue;
            }

            answer += (long) Math.floor(h2) - (long) Math.floor(h1);

            if (h1 % 1 == 0) {
                answer += 1;
            }
        }

        return 4 * (answer + 1);
    }
}